src/Microsoft.Framework.PackageManager/Install/InstallGlobalCommand.cs

```diff
@@ -40,67 +42,212 @@ public FeedOptions FeedOptions
+            var tempFolderFullPath = Path.Combine(installFolder, tempFolderName);
+
+            // Delete if exists already
+            FileOperationUtils.DeleteFolder(tempFolderFullPath);
+            Directory.CreateDirectory(tempFolderFullPath);
 
-            RestoreCommand.RestorePackageId = packageId;
-            RestoreCommand.RestorePackageVersion = packageVersion;
+            WriteVerbose("Temporary folder name: {0}", tempFolderFullPath);
+            var projectFileFullPath = Path.Combine(tempFolderFullPath, "project.json");
 
-            return
-                await RestoreCommand.ExecuteCommand() &&
-                InstallCommands(packageId, RestoreCommand.AppInstallPath);
+            File.WriteAllText(
+                projectFileFullPath,
+                string.Format(
+        @"{{
+    ""dependencies"":{{
+        ""{0}"":""{1}""
+    }}
+}}", packageName, packageVersion));
+
+            return projectFileFullPath;
         }
 
-        private bool InstallCommands(string packageId, string appPath)
+        private bool ValidateApplicationPackage(string appFolderFullPath)
         {
-            string commandsFolder = Path.Combine(appPath, InstallBuilder.CommandsFolderName);
+            string commandsFolder = Path.Combine(appFolderFullPath, InstallBuilder.CommandsFolderName);
 
             if (!Directory.Exists(commandsFolder))
             {
-                LogError("The specified package is not an application. The package was added but no commands were installed.");
+                WriteError("The specified package is not an application. The package was added but no commands were installed.");
                 return false;
             }
 
-            var fileFilter = PlatformHelper.IsWindows ? "*.cmd" : "*.sh";
-            var allAppCommandsFiles = Directory.EnumerateFiles(commandsFolder, fileFilter);
-            var allAppCommands = allAppCommandsFiles
-                .Select(cmd => Path.GetFileNameWithoutExtension(cmd))
-                .ToList();
-
-            var blockedCommands = _commandsRepository.Commands.Where(cmd => 
+            var blockedCommands = _commandsRepository.Commands.Where(cmd =>
                 !CommandNameValidator.IsCommandNameValid(cmd) ||
                 CommandNameValidator.ShouldNameBeSkipped(cmd));
             if (blockedCommands.Any())
             {
-                LogError(string.Format(
+                WriteError(string.Format(
                     "The application cannot be installed because the following command names are not allowed: {0}.",
                     string.Join(", ", blockedCommands)));
 
                 return false;
             }
 
+            return true;
+        }
+
+        private bool InstallCommands(string packageId, string appPath)
+        {
+            string commandsFolder = Path.Combine(appPath, InstallBuilder.CommandsFolderName);
+
+            var fileFilter = PlatformHelper.IsWindows ? "*.cmd" : "*.sh";
```

Comment:
BTW, we don't call the file .sh anywhere else. We should remove that. kpm bundle today makes foo.cmd and foo.